From 74c69da52ad5029cd2fc2d3d41d581022ed0a7aa Mon Sep 17 00:00:00 2001
From: Waldemar Brodkorb <wbx@openadk.org>
Date: Wed, 15 Apr 2015 12:47:57 -0500
Subject: [PATCH] fix static linking of pthread apps

When compiling python you get duplicate symbol problem.
Seen in the autobuilders of buildroot project.

Signed-off-by: Waldemar Brodkorb <wbx@uclibc-ng.org>
---
 libc/sysdeps/linux/common/setregid.c  | 2 +-
 libc/sysdeps/linux/common/setresgid.c | 4 ++--
 libc/sysdeps/linux/common/setresuid.c | 4 ++--
 3 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/libc/sysdeps/linux/common/setregid.c b/libc/sysdeps/linux/common/setregid.c
index 15d590a..aaa7293 100644
--- a/libc/sysdeps/linux/common/setregid.c
+++ b/libc/sysdeps/linux/common/setregid.c
@@ -37,4 +37,4 @@ int setregid(gid_t rgid, gid_t egid)
 }
 #endif
 
-libc_hidden_def(setregid)
+libc_hidden_weak(setregid)
diff --git a/libc/sysdeps/linux/common/setresgid.c b/libc/sysdeps/linux/common/setresgid.c
index 95decc2..b6d1647 100644
--- a/libc/sysdeps/linux/common/setresgid.c
+++ b/libc/sysdeps/linux/common/setresgid.c
@@ -16,7 +16,7 @@
 # define __NR_setresgid __NR_setresgid32
 
 _syscall3(int, setresgid, gid_t, rgid, gid_t, egid, gid_t, sgid)
-libc_hidden_def(setresgid)
+libc_hidden_weak(setresgid)
 
 #elif defined(__NR_setresgid)
 
@@ -34,7 +34,7 @@ int setresgid(gid_t rgid, gid_t egid, gid_t sgid)
 	}
 	return (__syscall_setresgid(rgid, egid, sgid));
 }
-libc_hidden_def(setresgid)
+libc_hidden_weak(setresgid)
 
 #endif
 
diff --git a/libc/sysdeps/linux/common/setresuid.c b/libc/sysdeps/linux/common/setresuid.c
index 022ccfe..a2a2183 100644
--- a/libc/sysdeps/linux/common/setresuid.c
+++ b/libc/sysdeps/linux/common/setresuid.c
@@ -16,7 +16,7 @@
 # define __NR_setresuid __NR_setresuid32
 
 _syscall3(int, setresuid, uid_t, ruid, uid_t, euid, uid_t, suid)
-libc_hidden_def(setresuid)
+libc_hidden_weak(setresuid)
 
 #elif defined(__NR_setresuid)
 
@@ -34,7 +34,7 @@ int setresuid(uid_t ruid, uid_t euid, uid_t suid)
 	}
 	return (__syscall_setresuid(ruid, euid, suid));
 }
-libc_hidden_def(setresuid)
+libc_hidden_weak(setresuid)
 
 #endif
 
-- 
2.3.2 (Apple Git-55)

